test/xor.ts

```diff
@@ -0,0 +1,26 @@
+import {expectType} from 'tsd-check';
+import {XOR} from '..';
+
+interface BaseOptions {
+	option?: string;
+}
+
+interface ExclusiveVariation1 extends BaseOptions {
+	exclusive1: boolean;
+}
+
+interface ExclusiveVariation2 extends BaseOptions {
+	exclusive2: number;
+}
+
+type Options = XOR<ExclusiveVariation1, ExclusiveVariation2>;
+
+const exclusiveVariation1: Options = {exclusive1: true};
+const exclusiveVariation2: Options = {exclusive2: 1};
+
+expectType<{option?: string; exclusive1: boolean; exclusive2?: string}>(
+	exclusiveVariation1
+);
+expectType<{option?: string; exclusive1?: string; exclusive2: number}>(
+	exclusiveVariation2
+);
```

Comment:
Can you add some commented-out negative tests.

So we can easily use those when https://github.com/SamVerschueren/tsd-check/issues/2 is fixed.

Example: https://github.com/sindresorhus/type-fest/pull/1/files#diff-fd2873d8135a3976c91e67f021a5cd40R21